apps/oxlint/src/lint.rs

```diff
@@ -291,6 +291,18 @@ impl CliRunner {
             }
         };
 
+        // TODO: refactor this elsewhere.
+        // This code is in the oxlint app, not in oxc_linter crate
+        if let Some(ref external_linter) = external_linter {
+            if let Err(err) = external_plugin_store.setup_configs(external_linter) {
+                print_and_flush_stdout(
+                    stdout,
+                    &format!("Failed to setup external plugin options: {err}\n"),
+                );
+                return CliRunResult::InvalidOptionConfig;
+            }
+        }
```

Comment:
I could use help here. I exposed `setup_configs()` for the `oxlint` crate by putting it on `ExternalPluginStore` only because it's the first place I found. I could do with a few pointers on separation of concerns regarding these structs.

Reply:
Yeah, the types are a bit of a convoluted mess here. I'm tied up today. Will take a look tomorrow.

Reply:
- Guessing this is related to https://github.com/oxc-project/oxc/issues/12437?

Reply:
I think the way you've done it is good!

Personally I find all the logic around config in linter labyrinthine and confusing/confused. You're asking about separation of concerns. The problem IMO is that there isn't any! Or at least if there is, it's not clear what it is (to me anyway).